sdk/onlineexperimentation/azure-onlineexperimentation/azure/onlineexperimentation/models/_models.py

```diff
@@ -0,0 +1,527 @@
+    """Diagnostic details for validation errors.
+
+    :ivar message: A human-readable error message. Required.
+    :vartype message: str
+    :ivar code: The diagnostic error code. Required. Known values are: "FailedSchemaValidation",
+     "InvalidEventCondition", "UnsupportedEventCondition", and "InvalidExperimentMetricDefinition".
+    :vartype code: str or ~azure.onlineexperimentation.models.DiagnosticCode
+    """
+
+    message: str = rest_field(visibility=["read"])
+    """A human-readable error message. Required."""
+    code: Union[str, "_models.DiagnosticCode"] = rest_field(visibility=["read"])
+    """The diagnostic error code. Required. Known values are: \"FailedSchemaValidation\",
+     \"InvalidEventCondition\", \"UnsupportedEventCondition\", and
+     \"InvalidExperimentMetricDefinition\"."""
+
+
+class EventCountMetricDefinition(ExperimentMetricDefinition, discriminator="EventCount"):
+    """The definition of an EventCount metric definition. Counts the occurrences of a specified event.
+
+    :ivar type: The type of metric. Required. Counts the occurrences of an event. Experiment
+     analysis accounts for unequal traffic allocation.
+    :vartype type: str or ~azure.onlineexperimentation.models.EVENT_COUNT
+    :ivar event: Event to observe. Required.
+    :vartype event: ~azure.onlineexperimentation.models.ObservedEvent
+    """
+
+    type: Literal[ExperimentMetricType.EVENT_COUNT] = rest_discriminator(name="type", visibility=["read", "create", "update", "delete", "query"])  # type: ignore
+    """The type of metric. Required. Counts the occurrences of an event. Experiment analysis accounts
+     for unequal traffic allocation."""
+    event: "_models.ObservedEvent" = rest_field(visibility=["read", "create", "update", "delete", "query"])
+    """Event to observe. Required."""
+
+    @overload
+    def __init__(
+        self,
+        *,
+        event: "_models.ObservedEvent",
+    ) -> None: ...
+
+    @overload
+    def __init__(self, mapping: Mapping[str, Any]) -> None:
+        """
+        :param mapping: raw JSON to initialize the model.
+        :type mapping: Mapping[str, Any]
+        """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, type=ExperimentMetricType.EVENT_COUNT, **kwargs)
+
+
+class EventRateMetricDefinition(ExperimentMetricDefinition, discriminator="EventRate"):
+    """The definition of an EventRate metric definition. Calculates the percentage of events
+    satisfying a specified condition.
+
+    :ivar type: The type of metric. Required. Calculates the percentage of events that satisfy a
+     specified condition.
+    :vartype type: str or ~azure.onlineexperimentation.models.EVENT_RATE
+    :ivar event: Event to observe as the rate denominator. Required.
+    :vartype event: ~azure.onlineexperimentation.models.ObservedEvent
+    :ivar rate_condition: The event contributes to the rate numerator if it satisfies this
```

Comment:
Same question for this field, what are the possible values?

Reply:
Same as the other one, this is a KQL expression based on the customer's event schema. Samples are better suited to adequately demonstrate the usage.